content-elements/form/form-pin/form-pin.js

```diff
@@ -0,0 +1,179 @@
+import Alpine from '@alpinejs/csp';
+import { Tooltip } from 'bootstrap';
+
+Alpine.data('formPin', () => ({
+  bsiInputElement: null, // Input field required for CX story / value flow
+  maxlength: null,
+  requiredErrorElement: null,
+  form: null,
+  root: null,
+
+  initForm() {
+    this.form = this.$el;
+    this.root = this.$root;
+
+    if (this.root.classList.contains('bsi-form-label-floating')) {
+      for(const floatingElement of this.form.getElementsByClassName('bsi-label-floating-element')) {
+        this._initFloatingLabels(floatingElement);
+      }
+    }
+
+    if (this.root.classList.contains('bsi-form-info-as-tooltip')
+        && ['bsi-form-label-top', 'bsi-form-label-left'].some(labelClass => this.root.classList.contains(labelClass))) {
+      this.form.querySelectorAll('.bsi-form-element').forEach((formElement) => {
+        let infoTextField = formElement.querySelector('.form-text');
+        let tooltipIcon = formElement.querySelector('i');
+        if (infoTextField && infoTextField.innerText && tooltipIcon) {
+          tooltipIcon.classList.add('tooltip-visible');
+          tooltipIcon.parentElement.classList.add('contains-tooltip');
+          tooltipIcon.setAttribute('title', infoTextField.innerText);
+          new Tooltip(tooltipIcon);
+        }
+      })
+    }
+  },
+
+  submitForm(e) {
+    if (!this.form.checkValidity()) {
+      e.preventDefault();
+      e.stopPropagation();
+      this.validateInput();
+    }
+    this.form.classList.add('was-validated');
+  },
+
+  initRequiredError() {
+    this.requiredErrorElement = this.$el;
+  },
+
+  validateInput() {
+    if (this.bsiInputElement.value.length != this.maxlength) {
+      this.requiredErrorElement.style.display = "block";
+    } else {
+      this.requiredErrorElement.style.display = "none";
+    }
+  },
+
+  initPinNumberFields() {
+    this.bsiInputElement = this.$root.querySelector('.bsi-form-field-input-original');
+    let containerDiv = this.$root.querySelector('.bsi-form-pin-element');
+
+    const maxLength = this.bsiInputElement.getAttribute('maxlength');
+    if (maxLength == null) {
+      this.maxlength = 6;
+    } else {
+      this.maxlength = this.bsiInputElement.getAttribute('maxlength');
+    }
```

Comment:
Das kannst du auch so schreiben:
`var maxLength = this.bsiInputElement.getAttribute('maxlength') ?? 6;`